### DISCUSSION TITLE
Recovering Rotation Axis From JointModelRevoluteUnaligned

### DISCUSSION THREAD

**Discussion by Cfather** [stack-of-tasks/pinocchio#2365] [Q&A] [answered]
Hi, I have a specific coding question. 

I have some revolute joints that rotate reversely in the robot. The urdf description for these joints would be something like `<axis="0 0 -1">`.
My question is: how could I get the information of what the rotation axis is from a pinocchio model.
For example, joints like `<axis="0 0 1">` will be parsed as `JointModelRZ`. And if I print the name of the joint using `model.joints[i].shortname()`, it will give me a string `JointModelRZ` and I will be able to know that this is a revolute joint with respect to z axis. 
Joints like `<axis="0 0 -1">` however, will be parsed as `JointModelRevoluteUnaligned` so I can't tell from its shortname which axis it rotates. 

I noticed that `JointModelRevoluteUnalignedTpl` contains a vector variable `axis` that might store this information. However, `model.joints[i]` stores a `JointModelTpl`. Is there a way to convert from `JointModelTpl` to `JointModelRevoluteUnalignedTpl` so that I can get access to the variable `axis` and then tell the rotation axis of this joint? I appreciate any ideas!

**Comment by jorisv** [comment, marked as answer]
Hello @Cfather,

This pull request https://github.com/stack-of-tasks/pinocchio/pull/2315/ have added the getMotionAxis method on joint model with aligned axis.

`JointModelTpl` inherit from a boost::variant that contains the implemented joint.
You can create boost::static_visitor that will call getMotionAxis for aligned joint and axis for unaligned joint.

**Reply by Cfather** [reply]
Thanks! Will this appear in future versions of Pinocchio?

**Reply by jorisv** [reply]
It's already available in Pinocchio 3.2.0.
